custom/icds_reports/utils/aggregation_helpers/distributed/agg_service_delivery_report.py

```diff
@@ -0,0 +1,256 @@
+            tmp_tablename=self.staging_tablename,
+        ), {}
+
+        yield """
+        DROP TABLE temp_ccs_table;
+        """, {}
+
+    def rollup_query(self, aggregation_level):
+        columns = (
+            ('state_id', 'state_id'),
+            ('district_id', lambda col: col if aggregation_level > 1 else "'All'"),
+            ('block_id', lambda col: col if aggregation_level > 2 else "'All'"),
+            ('supervisor_id', lambda col: col if aggregation_level > 3 else "'All'"),
+            ('awc_id', lambda col: col if aggregation_level > 4 else "'All'"),
+            ('aggregation_level', str(aggregation_level)),
+            ('month', 'month'),
+            ('pse_eligible',),
+            ('pse_0_days',),
+            ('pse_1_7_days',),
+            ('pse_8_14_days',),
+            ('pse_15_20_days',),
+            ('pse_21_days',),
+            ('thr_eligible',),
+            ('thr_0_days',),
+            ('thr_1_7_days',),
+            ('thr_8_14_days',),
+            ('thr_15_20_days',),
+            ('thr_21_days',),
+            ('lunch_eligible',),
+            ('lunch_0_days',),
+            ('lunch_1_7_days',),
+            ('lunch_8_14_days',),
+            ('lunch_15_20_days',),
+            ('lunch_21_days',),
+            ('state_is_test', 'MAX(state_is_test)'),
+            (
+                'district_is_test',
+                lambda col: 'MAX({column})'.format(column=col) if aggregation_level > 1 else "0"
+            ),
+            (
+                'block_is_test',
+                lambda col: 'MAX({column})'.format(column=col) if aggregation_level > 2 else "0"
+            ),
+            (
+                'supervisor_is_test',
+                lambda col: 'MAX({column})'.format(column=col) if aggregation_level > 3 else "0"
+            ),
+            (
+                'awc_is_test',
+                lambda col: 'MAX({column})'.format(column=col) if aggregation_level > 4 else "0"
+            )
+        )
+        def _transform_column(column_tuple):
+            column = column_tuple[0]
+
+            if len(column_tuple) == 2:
+                agg_col = column_tuple[1]
+                if isinstance(agg_col, str):
+                    return column_tuple
+                elif callable(agg_col):
+                    return (column, agg_col(column))
+            return (column, 'SUM({})'.format(column))
+
+        columns = list(map(_transform_column, columns))
+
+        group_by = ["state_id"]
+        child_location = 'district_is_test'
+        if aggregation_level > 1:
+            group_by.append("district_id")
+            child_location = 'block_is_test'
+        if aggregation_level > 2:
+            group_by.append("block_id")
+            child_location = 'supervisor_is_test'
+        if aggregation_level > 3:
+            group_by.append("supervisor_id")
+            child_location = 'awc_is_test'
+
+        group_by.append('month')
+        group_by = ", ".join(group_by)
```

Comment:
nit: i prefer changing the variable name i the datatype is changing. so in this case i might use `group_by_text`, even though `group_by` is no longer needed